File: array/max_min.cpp
// #include<iostream>
// using namespace std;

// int main(){
//  int arr[10];
//  cout<<"Enter the value in array";
//  for(int i=0; i<10; i++)
//  {
//     cin>>arr[i];
//  }
//  for (int i = 0; i < 10; i++)
//  {
//     cout<<"value of Array is"<<arr[i]<<endl;
//  }
 
// }




#include<iostream>
using namespace std;


void findMinValueInArray(int array[]){

    int minnumber;
    minnumber=array[0];
    for (int i = 0; i <=2; i++)
    {
       
        if(minnumber>=array[i]){
           minnumber=array[i];
        } 
    }

    cout<<"the minimum number is here " << minnumber <<endl;
    

}


void findMixValueInArray(int arr[])
{
    int maxvalue;
    maxvalue=arr[0];
    for (int i = 0; i <= 2; i++)
    {
        if (maxvalue<=arr[i])
        {
           maxvalue=arr[i];
        }
    }

    cout<<"the max value is "<< maxvalue<<endl;
      

}

int main(){
 int arr[10];
 cout<<"Enter the value in array";
 for(int i=0; i<3; i++)
 {
    cin>>arr[i];
 }

 
 findMinValueInArray(arr);
 findMixValueInArray(arr);

}

File: array/max_min.dart
import 'dart:io';

void findMinValueInList(List<int> numbers) {
  int minNumber = numbers[0]; 

  for (int i = 1; i < numbers.length; i++) {
    if (numbers[i] < minNumber) {
      minNumber = numbers[i];
    }
  }
  print("The minimum number is: $minNumber");
}

void findMaxValueInList(List<int> numbers) {
  int maxNumber = numbers[0]; 
  for (int i = 1; i < numbers.length; i++) {
    if (numbers[i] > maxNumber) {
      maxNumber = numbers[i];
    }
  }

  print("The maximum number is: $maxNumber");
}

void main() {
  List<int> numbers = [];
  for (int i = 0; i < 3; i++) {
    String? inputString = stdin.readLineSync();
    int? input = int.tryParse(inputString?.trim() ?? ''); 
    if (input != null) {
      numbers.add(input);
    } else {
      print("Invalid input, please enter a number.");
      i--; 
    }
  }

  findMinValueInList(numbers);
  findMaxValueInList(numbers);
}
